### DISCUSSION TITLE
Character teleporting out of the ship while driving

### DISCUSSION THREAD

**Discussion by Vykn** [FakeFishGames/Barotrauma#14118] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Occasionally, while driving, the character will briefly teleport out of the sub, in the same way that would sometimes happen when moving between hulls. It might not only occur while driving, but since I spent 90% of my recent campaign driving, it's the only time I remember experiencing it. It happened maybe once every 10-20 minutes or so.

I've only ever experienced this in this Unstable, but I've heard it has happened in previous versions too. Perhaps it only happens when hosting from one's own computer.

https://github.com/FakeFishGames/Barotrauma/assets/37850646/b9b94743-8cae-4244-9039-e5fe4fe05694

### Reproduction steps

1. Load unstable campaign (you can use the save file I uploaded [here](https://github.com/FakeFishGames/Barotrauma/discussions/14117) if you want
2. Drive around (try to go fast? I think it might have happened more often as I drove faster)
3. Observe your character going to a different dimension sometimes

### Bug prevalence

Happens regularly

### Single player or multiplayer?

Multiplayer (Hosted from in-game menu)

### Version

v1.5.4.0 (unstable)

**Comment by A2ur3** [comment]
It happens in current public version. most of our server players have experienced it
a video shared by another player

https://github.com/FakeFishGames/Barotrauma/assets/64390906/e3113dab-08d1-4417-b311-f706e9d53ccb

**Comment by A2ur3** [comment]
this issue still happens after summer update

**Reply by A2ur3** [reply]
this issue still happens after autumn update

**Comment by lIllIIlI** [comment]
This has been happening for a long time, and happens every round, for the last 1000hr of my play time. 
No response from devs to this day huh. 

I had reported a worse version of this problem too here:
https://github.com/FakeFishGames/Barotrauma/discussions/14585

If the game takes too long to return the character/camera to the sub, they are permanently stuck outside and desyncd until relog, sometimes it results in deletion of the character itself if you die in the void due to poison or other afflictions.

**Comment by Jasontti** [comment]
Tested a fix for this.
Character stayed inside the sub even when server was under heavy load.
Closing discussion as resolved for now.

**Comment by lIllIIlI** [comment]
I don't see any commits linked. Can we not close discussions without confirmation?

**Reply by Jasontti** [reply]
Commits are not visible on the public git repo, so we can't link them here.
They can be searched with discord bot or i can paste them here if you want.
It is best that we close the discussions when we are done with ticket.

**Reply by lIllIIlI** [reply]
Usually Regalis posts the commit link in relevant discussions, or maybe only in issues.
For tracking purposes

**Reply by Jasontti** [reply]
Ah yes. Those are on the ticket.
In some cases it might have been forgotten, so ask about it if necessary.
https://github.com/FakeFishGames/Barotrauma/issues/15138#issuecomment-2473661948
